tools/src/commands/EasDispatch.ts

```diff
@@ -0,0 +1,176 @@
+import { Command } from '@expo/commander';
+import spawnAsync from '@expo/spawn-async';
+import assert from 'assert';
+import fs, { mkdirp } from 'fs-extra';
+import glob from 'glob-promise';
+import path from 'path';
+
+import { EXPO_DIR } from '../Constants';
+import logger from '../Logger';
+
+const RELEASE_BUILD_PROFILE = 'release-client';
+
+const CUSTOM_ACTIONS = {
+  'ios-client-build-and-submit': {
+    name: 'Build a new iOS client and submit it to the App Store',
+    actionId: 'ios-client-build-and-submit',
+    action: iosBuildAndSubmitAsync,
+  },
+  'android-client-build-and-submit': {
+    name: 'Build a new Android client and submit it to the Play Store',
+    actionId: 'android-client-build-and-submit',
+    action: androidBuildAndSubmitAsync,
+  },
+};
+
+export default (program: Command) => {
+  program
+    .command('eas-dispatch [action]')
+    .alias('eas')
+    .description(`Runs predefined EAS Build & Submit jobs.`)
+    .asyncAction(main);
+};
+
+async function main(actionId: string | undefined) {
+  if (!actionId || !CUSTOM_ACTIONS[actionId]) {
+    const actions = Object.values(CUSTOM_ACTIONS).map((i) => `\n- ${i.actionId} - ${i.name}`);
+    if (!actionId) {
+      logger.error(`You need to provide action name. Select one of: ${actions}`);
+    } else {
+      logger.error(`Unknown action ${actionId}. Select one of: ${actions}`);
+    }
+    return;
+  }
+
+  CUSTOM_ACTIONS[actionId].action();
+}
+
+async function iosBuildAndSubmitAsync() {
+  const projectDir = path.join(EXPO_DIR, 'apps/eas-expo-go');
+  const credentialsDir = path.join(projectDir, 'credentials');
+  const fastlaneMatchBucketCopyPath = path.join(credentialsDir, 'expo-client-certificates');
+  await mkdirp(fastlaneMatchBucketCopyPath);
+  await spawnAsync('gsutil', [
+    'rsync',
+    '-r',
+    '-d',
+    'gs://expo-client-certificates',
```

Comment:
nit: though it's not really critical. if we don't like exposing the bucket name, could we store the bucket name inside `secrets/keys.json` and let git-crypt help to do basic security for us?

Reply:
this is already public in fastlane/Matchfile